mlir/include/mlir/IR/BuiltinTypes.h

```diff
@@ -39,6 +39,11 @@ struct IntegerTypeStorage;
 struct TupleTypeStorage;
 } // namespace detail
 
+/// Type trait indicating that the type can be an operand to an elementwise op.
```

Comment:
This description, as well as the name of the trait does not seem to be something that we couldn't apply to Memref. 
However this isn't the intent right?

Reply:
No, it is not the intent. How would you phrase it better? I want this trait to be for value types like vector or tensor, but not for memrefs. MemRefs are just pointers, it is not clear what we actually map.

Reply:
Memref are more than pointers: they have a shape, which means you can actually iterate all the elements inside a memref. This makes it a candidate for a "map" function I believe (I understand this isn't what you want, which is why I'm picking on the name).
I think @jpienaar proposed something for this?

Reply:
`MappableValueBasedContainer`?

Reply:
Isn't this conflating two different traits? For an `arith` op on a shaped type you want it to have value semantics and you want the right element type. Why does that need to be expressed in a single trait?

Reply:
Added the second trait.